apps/dashr-tsne/app.R

```diff
@@ -0,0 +1,1195 @@
+library(dplyr)
+library(dash)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(data.table)
+library(plotly)
+library(rlist)
+library(glue)
+library(parallel)
+library(compiler)
+library(dtplyr)
+library(knitr)
+library(png)
+library(magick)
+library(base64enc)
+library(magrittr)
+library(Rtsne)
+library(stringr)
+library(jsonlite)
+library(tictoc)
+
+appName <- Sys.getenv("DASH_APP_NAME")
+pathPrefix <- sprintf("/%s/", appName)
+
+
+Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+           DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+setwd(sprintf("/app/apps/%s", appName))
+
+l. <- cmpfun(list)
+
+IMAGE_DATASETS <- c("mnist_3000", "cifar_gray_3000", "fashion_3000")
+WORD_EMBEDDINGS <- c("wikipedia_3000", "twitter_3000", "crawler_3000")
+
+
+readme <- htmlDiv(children=l.(
+  htmlButton("Close", id='close-button', n_clicks=0)
+  ,
+  dccMarkdown("# t-SNE Explorer
+
+This is a demo of the Dash interactive R framework developed by [Plotly](https://plot.ly/).
+
+Dash abstracts away all of the technologies and protocols required to build an interactive web-based application and is a simple and effective way to bind a user interface around your Python code. To learn more check out our [documentation](https://plot.ly/dash). 
+
+For an introductory and extensive explanation of t-SNE how to use it properly, please check out the [demo app](https://dash-tsne.plot.ly/).
+")
+  ,
+  htmlImg(id='gif1', src='assets/animated1.gif')
+  ,
+  htmlImg(id='gif2', src='assets/animated2.gif')
+  ,
+  dccMarkdown("
+## Getting Started
+### Using the demo
+To get started, choose a dataset you want to visualize. When the scatter plot appears on the graph, you can see the original image by clicking on a data point. 
+
+Alternatively, you can explore the GloVe Word Vectors datasets, which are encoded vectors of large collection of texts from Wikipedia, Twitter, and acquired through Web Crawlers. Upon clicking a data point, you will be able to see the 5 closest neighbors of the word you clicked.
+
+### Running the app locally
+
+
+
+Clone the git repo, then install the requirements with pip
+```
+git clone https://github.com/plotly/dashr-tsne.git
+cd dashr-tsne
+
+```
+Download the script and run init.R
+
+```
+Rscript init.R
+```
+Run the app
+```
+Rscript init.R
+```
+or using ctrl+shift+enter (cmd+shift+enter)
+### How to use the local version
+To train your own t-SNE algorithm, input a high-dimensional dataset with only numerical values, and the corresponding labels inside the upload fields. For convenience, small sample datasets are included inside the data directory. [You can also download them here](https://www.dropbox.com/sh/l79mcmlqil7w7so/AACfQhp7lUS90sZUedsqAzWOa?dl=0&lst=). The training can take a lot of time depending on the size of the dataset (the complete MNIST dataset could take 15-30 min), and it is not advised to refresh the webpage when you are doing so.
```

Comment:
Yikes! :hocho: Dropbox references please. If they're in the original Python version, they should be :hocho: there also (Sham can help with that if so).